### PR TITLE
fix(build): Support loading in CommonJS environments like Node.

### LINKED ISSUE #25: Update to be compatible with CommonJS
The relier client is going to be used on the server to trade codes for tokens, then work with user profiles. Update FxaRelierClient.js to be compatibile with the CommonJS module system.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
fixes #25

### CODE REVIEW

**Comment by vladikoff** [comment]
I am not a fan of the minified module wrapper, it makes it harder to debug. (we minify the contents in the end anyway). In addition for the fxa-js-client we have an amd loader that allows us to publish the module for npm: https://github.com/mozilla/fxa-js-client/blob/master/node/amd-loader.js

**Comment by shane-tomlinson** [comment]
> I am not a fan of the minified module wrapper, it makes it harder to debug. 

I can un-minify the wrapper pretty easily. Still, more interesting is the `amd-loader.js` you linked to.

**Comment by vladikoff** [comment]
Probably a better file to link to https://github.com/mozilla/fxa-js-client/blob/master/node/index.js

**Comment by shane-tomlinson** [comment]
@vladikoff - updated with your suggestions, much nicer. Mind looking over it when you get a few?

File: node/amd-loader.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    exported = factory.apply(null, modules);
+  };
+  define.amd = true;
+  var amdrequire = function amdrequire(filepath) {
+    // Return real node modules if we have them mapped
+    if (filepath in map) {
+      return require(map[filepath]);
+    }
+
+    // be default, cwd is the baseUrl unless the filepath
```
Comment (shane-tomlinson) [inline node/amd-loader.js]: `be default`->`by default`

**Comment by vladikoff** [comment]
Works well!
image

**Pull request merged**